helm/cdash/templates/cdash/secret.yaml

```diff
@@ -0,0 +1,55 @@
+{{- $appSecretName := printf "%s-%s" .Release.Name "website" -}}
+---
+kind: Secret
+apiVersion: v1
+metadata:
+  name: {{ $appSecretName }}
+  annotations:
+    "helm.sh/resource-policy": "keep"
+type: Opaque
+data:
+  # Generate a random APP_KEY if one doesn't already exist
+  {{- $oldAppSecret := lookup "v1" "Secret" .Release.Namespace $appSecretName }}
+  {{- $oldAppSecretData := (get $oldAppSecret "data") | default dict }}
+  APP_KEY: {{ (get $oldAppSecretData "APP_KEY") | default (randAlphaNum 32 | b64enc) | quote }}
```

Comment:
Is there an easy way to reuse this so we don't have to duplicate similar token-generation logic below?

Reply:
I agree, that solution is probably less readable than your current solution.  The current solution is probably good enough if there isn't an obviously cleaner way to do it.